Workspaces/Inspector/Scripts/InspectorListViewController.cs

```diff
@@ -205,8 +214,8 @@ protected override InspectorListItem GetItem(InspectorData listData)
 			if (headerItem)
 			{
 				var go = (GameObject)listData.serializedObject.targetObject;
-				headerItem.lockToggle.isOn = isLocked(go);
 				headerItem.setLocked = locked => setLocked(go, locked);
+				headerItem.lockToggle.isOn = isLocked(go);
```

Comment:
Fix for bug with switching objects in the HierarchyWorkspace would unset a lock on a previous game object (from the closure). Setting the callback first fixes this.